react-native-gutenberg-bridge/ios/GutenbergWebFallback/GutenbergWebSingleBlockViewController.swift

```diff
@@ -0,0 +1,178 @@
+         }
+     }
+
+    private func loadWebView() {
+        getRequest(for: webView) { [weak self] (request) in
+            self?.webView.load(request)
+        }
+    }
+
+    @objc public func onSaveButtonPressed() {
+        evaluateJavascript(jsInjection.getHtmlContentScript)
+    }
+
+    @objc public func onCloseButtonPressed() {
+        delegate?.webControllerDidPressClose(controller: self)
+    }
+
+    private func addNavigationBarElements() {
+        let saveButton = UIBarButtonItem(barButtonSystemItem: .save, target: self, action: #selector(onSaveButtonPressed))
+        let cancelButton = UIBarButtonItem(barButtonSystemItem: .cancel, target: self, action: #selector(onCloseButtonPressed))
+
+        navigationItem.rightBarButtonItem = saveButton
+        navigationItem.leftBarButtonItem = cancelButton
+        let localizedTitle = NSLocalizedString("Edit %@ block", comment: "Title of Gutenberg WEB editor running on a Web View. %@ is the localized block name.")
+        title = String(format: localizedTitle, block.name)
+    }
+
+    func addCoverView() {
+        webView.addSubview(coverView)
+        if #available(iOS 13.0, *) {
+            coverView.backgroundColor = UIColor.systemBackground
+        } else {
+            coverView.backgroundColor = .white
+        }
+        coverView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            coverView.leadingAnchor.constraint(equalTo: webView.leadingAnchor),
+            coverView.trailingAnchor.constraint(equalTo: webView.trailingAnchor),
+            coverView.topAnchor.constraint(equalTo: webView.topAnchor),
+            coverView.bottomAnchor.constraint(equalTo: webView.bottomAnchor),
+        ])
+    }
+
+    func removeCoverViewAnimated() {
+        UIView.animate(withDuration: 1, animations: {
+            self.coverView.alpha = 0
+        }) { _ in
+            self.coverView.removeFromSuperview()
+            self.coverView.alpha = 1
+        }
+    }
+
+    private func evaluateJavascript(_ script: WKUserScript) {
+        webView.evaluateJavaScript(script.source) { (response, error) in
+            if let response = response {
+                print("\(response)")
+            }
+            if let error = error {
+                print("\(error)")
+            }
+        }
+    }
+
+    private func save(_ newContent: String) {
+        delegate?.webController(controller: self, didPressSave: block.replacingContent(with: newContent))
+    }
+}
+
+extension GutenbergWebSingleBlockViewController: WKNavigationDelegate {
+    public func webView(_ webView: WKWebView, decidePolicyFor navigationResponse: WKNavigationResponse, decisionHandler: @escaping (WKNavigationResponsePolicy) -> Void) {
+        if !isWPOrg && navigationResponse.response.url?.absoluteString.contains("/wp-admin/post-new.php") ?? false {
+            evaluateJavascript(jsInjection.insertBlockScript)
```

Comment:
After the main `new-post.php` request returns, WPCom won't load gutenberg until it returns a different request to `https://widgets.wp.com/notifications/?jetpack=true&v=@automattic/jetpack-blocks@13.1.0-5015-gd57f427480&locale=en#SITE_URL/wp-admin/post-new.php`

Trying to insert before this request returns will do nothing.

In the other hand, waiting for the main request to return on `didRecievePolicy` on WPOrg sites won't work since it happens just before `onLoad`, and there are no other pending requests.

We would be in troubles if the requests flow changes in WPCom, and I'm open to suggestions :)

And this reminded me that we will need to test WPOrg sites with Jetpack. 🤔

Reply:
I think I'll have to revisit your above comment when testing again on a .org site to understand things better, but thanks for the explanation!